cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetCache.java

```diff
@@ -129,6 +124,11 @@ public DynamicDatasetCache(SystemDatasetInstantiator instantiator,
     }
   }
 
+  @Override
+  public void releaseDataset(Dataset dataset) {
+    dismissDataset(dataset);
```

Comment:
So the current implementation is not to reuse a dataset instance once it is released?